dbt/include/spark/macros/materializations/snapshot.sql

```diff
@@ -81,13 +82,12 @@
 
 
 {% materialization snapshot, adapter='spark' %}
-  {%- set config = model['config'] -%}
 
   {%- set target_table = model.get('alias', model.get('name')) -%}
 
   {%- set strategy_name = config.get('strategy') -%}
   {%- set unique_key = config.get('unique_key') %}
-  {%- set file_format = config.get('file_format', 'parquet') -%}
+  {%- set file_format = config.get('file_format') or 'parquet' -%}
```

Comment:
Isn't this the same thing? Or does python return `None` here if `config["file_format"] == None`?

Reply:
I switched from using the dictionary version of config to using the config object in the context, and they require different ways of specifying defaults.

Reply:
With a dictionary the `config.get('file_format', 'parquet') works and returns "parquet" if file_format is None. With the config object, if file_format is None, you don't get "parquet", you have to do it outside of the "get" call. (This is because of the way we've implemented "get" in the BaseConfig object. And it has always felt like it would break too much to change the behavior.)

Reply:
It would be nice to comment somewhere that the get behavior has been overloaded such that this is required. `or` is considered a Python antipattern over `get` often, so it'd be nice to make it known why we do this.

Reply:
Comments have been added.